<?xml version="1.0"?>

<launch>
    <arg name="pub_map_odom_transform" default="true"/> 
    <arg name="map_frame" default="map"/> 
    <arg name="base_frame" default="base_link"/> 
    <arg name="odom_frame" default="odom"/>
    <arg name="scan_topic" default="lidars/front_lidar/scan"/>
    <arg name="map_size" default="3072"/> <!-- range 150m --> <!-- 1024 <-> 50m -->

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigation2d_example)/rviz_cfg/rviz_cfg.rviz"/>

    <include file="$(find navigation2d_example)/launch/default_mapping.launch"> 
	<arg name="base_frame" value="$(arg base_frame)"/>
	<arg name="odom_frame" value="$(arg base_frame)"/>
	<arg name="scan_topic" value="$(arg scan_topic)"/>
        <arg name="map_size" value="$(arg map_size)"/>
    </include>

    <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch">
	<arg name="trajectory_source_frame_name" value="$(arg base_frame)"/>
    </include>
</launch>
